packages/core/src/components/graphs/bar-simple.ts

```diff
@@ -47,26 +47,62 @@ export class SimpleBar extends Bar {
 		const barsEnter = bars.enter()
 			.append("rect")
 			.attr("opacity", 0);
 
-		barsEnter.merge(bars)
-			.classed("bar", true)
-			.attr("x", (d, i) => {
-				const barWidth = this.getBarWidth();
-
-				return this.services.axes.getXValue(d, i) - barWidth / 2;
-			})
-			.attr("width", this.getBarWidth.bind(this))
-			.transition(this.services.transitions.getTransition("bar-update-enter", animate))
-			.attr("y", (d, i) => this.services.axes.getYValue(Math.max(0, d.value)))
-			.attr("fill", d => this.model.getFillScale()(d.label))
-			.attr("height", (d, i) => {
-				return Math.abs(this.services.axes.getYValue(d, i) - this.services.axes.getYValue(0));
-			})
-			.attr("opacity", 1);
+		// the mapping of rect attributes switches for horizontal vs vertical bar charts
+		if (this.model.getOptions().orientation === BarOrientationOptions.VERTICAL) {
+			// vertical bar charts needs the main Y axis for bar lengths
+			const yScale = this.services.axes.getMainYAxis();
+			barsEnter.merge(bars)
+				.classed("bar", true)
+				.attr("x", (d, i) => {
+					const barWidth = this.getBarWidth();
+					return this.services.axes.getXValue(d, i) - barWidth / 2;
+				})
+				.attr("width", this.getBarWidth.bind(this))
+				.transition(this.services.transitions.getTransition("bar-update-enter", animate))
+				.attr("y", (d, i) => {
+					if (yScale.scaleType === ScaleTypes.LABELS) {
+						return this.services.axes.getYValue(d.label, i);
+					} else {
+						return this.services.axes.getYValue(Math.max(this.services.axes.getYValue(0), d.value));
+					}
+				})
+				.attr("fill", d => this.model.getFillScale()(d.label))
+				.attr("height", (d, i) => {
+					if (yScale.scaleType === ScaleTypes.LABELS) {
+						return Math.abs(this.services.axes.getYValue(d, i) - yScale.scale.range()[0]); }
+
+					return Math.abs(this.services.axes.getYValue(d, i) - this.services.axes.getYValue(0));
+				})
+				.attr("opacity", 1);
+
+		} else {
+			// horizontal bars depend on the main X axis for length
+			const xScale = this.services.axes.getMainXAxis();
+			barsEnter.merge(bars)
+				.classed("bar", true)
+				.attr("x", (d, i) => {
+					if (xScale.scaleType === ScaleTypes.LABELS) {
+						return xScale.scale.range()[0];
+					}
+					return d.value > 0 ? this.services.axes.getXValue(0) : this.services.axes.getXValue(d, i);
+				})
+				.attr("width", (d, i) => {
+					if (xScale.scaleType === ScaleTypes.LABELS) {
+						return Math.abs(this.services.axes.getXValue(d.label, i) - xScale.scale.range()[1]);
+					}
+					return Math.abs(this.services.axes.getXValue(d, i) - this.services.axes.getXValue(0));
+				})
+				.transition(this.services.transitions.getTransition("bar-update-enter", animate))
+				.attr("y", (d, i) => this.services.axes.getYValue(d, i) - (this.getBarWidth() / 2))
+				.attr("fill", d => this.model.getFillScale()(d.label))
+				.attr("height", this.getBarWidth.bind(this))
+				.attr("opacity", 1);
+		}
```

Comment:
It'd be nice if there was a way to reduce the level of code duplication ... Maybe pulling out some of the functions would help?